src/server/address_manager_store.py

```diff
@@ -0,0 +1,214 @@
+            "CREATE TABLE IF NOT EXISTS new_table("
+            "node_id int,"
+            "bucket int)"
+        )
+        await self.db.commit()
+        return self
+
+    async def clear(self):
+        cursor = await self.db.execute("DELETE from metadata")
+        await cursor.close()
+        await self.db.commit()
+        cursor = await self.db.execute("DELETE from nodes")
+        await cursor.close()
+        await self.db.commit()
+        cursor = await self.db.execute("DELETE from new_table")
+        await cursor.close()
+        await self.db.commit()
+
+    async def get_metadata(self):
+        cursor = await self.db.execute("SELECT key, value from metadata")
+        metadata = await cursor.fetchall()
+        await cursor.close()
+        return {key: value for key, value in metadata}
+
+    async def is_empty(self):
+        metadata = await self.get_metadata()
+        if "key" not in metadata:
+            return True
+        if (
+            "new_count" in metadata
+            and metadata["new_count"] > 0
+        ):
+            return False
+        if (
+            "tried_count" in metadata
+            and metadata["tried_count"] > 0
+        ):
+            return False
+        return True
+
+    async def get_nodes(self):
+        cursor = await self.db.execute("SELECT node_id, value from nodes")
+        nodes_id = await cursor.fetchall()
+        await cursor.close()
+        return [
+            (node_id, ExtendedPeerInfo.from_string(info_str))
+            for node_id, info_str in nodes_id
+        ]
+
+    async def get_new_table(self):
+        cursor = await self.db.execute("SELECT node_id, bucket from new_table")
+        entries = await cursor.fetchall()
+        await cursor.close()
+        return [(node_id, bucket) for node_id, bucket in entries]
+
+    async def set_metadata(self, metadata):
+        for key, value in metadata:
+            cursor = await self.db.execute(
+                "INSERT OR REPLACE INTO metadata VALUES(?, ?)",
+                (key, value),
+            )
+            await cursor.close()
+            await self.db.commit()
+
+    async def set_nodes(self, node_list):
+        for node_id, peer_info in node_list:
+            cursor = await self.db.execute(
+                "INSERT OR REPLACE INTO nodes VALUES(?, ?)",
+                (node_id, peer_info.to_string())
+            )
+            await cursor.close()
+            await self.db.commit()
+
+    async def set_new_table(self, entries):
+        for node_id, bucket in entries:
+            cursor = await self.db.execute(
+                "INSERT OR REPLACE INTO new_table VALUES(?, ?)",
+                (node_id, bucket),
+            )
+            await cursor.close()
+            await self.db.commit()
+
+    async def serialize(self, address_manager: AddressManager):
+        metadata = []
+        nodes = []
+        new_table_entries = []
+
+        metadata.append(("key", str(address_manager.key)))
```

Comment:
Why are we using this format as opposed to Streamable or sqlite columns? please add a comment